core/src/main/scala/com/criteo/cuttle/Executor.scala

```diff
@@ -827,16 +847,29 @@ class Executor[S <: Scheduling] private[cuttle] (val platforms: Seq[ExecutionPla
                 .toLeft {
                   val nextExecutionId = utils.randomUUID
 
+                  val streams = new ExecutionStreams {
+                    def writeln(str: CharSequence) = ExecutionStreams.writeln(nextExecutionId, str)
+                  }
+
+                  // wrap the execution context so that we can register the name of the thread of each
+                  // runnable (and thus future) that will be run by the side effect.
+                  val sideEffectExecutionContext = SideEffectExecutionContext.wrap(runnable => new Runnable {
+                      override def run(): Unit = {
+                        val tName = Thread.currentThread().getName
+                        Executor.threadNamesToStreams.put(tName, streams)
+                        runnable.run()
```

Comment:
Just adding a `try { ... } finally { ... }` block here is enough to always clean the map and to not have to think about potential memory leaks.

Reply:
I  am not sure this would be sufficient, I am mostly tthinking of InterruptedException, as the runnable handling the registration/unregistration itself is also subject to being interrupted as well as the runnable being wrapped around which we would catch the exception.
Granted this is an edge case and should likely not be an issue

Reply:
I updated the code